src/ResourceManager/KeyVault/Commands.KeyVault/Commands/ManagedStorageAccounts/SetAzureKeyVaultManagedStorageAccountAttribute.cs

```diff
@@ -0,0 +1,107 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.Commands.KeyVault.Models;
+using System;
+using System.Collections;
+using System.Management.Automation;
+
+namespace Microsoft.Azure.Commands.KeyVault
+{
+    [Cmdlet( VerbsCommon.Set, CmdletNoun.AzureKeyVaultManagedStorageAccountAttribute,
+        SupportsShouldProcess = true,
+        HelpUri = Constants.KeyVaultHelpUri )]
+    [OutputType( typeof( ManagedStorageAccount ) )]
+    public class SetAzureKeyVaultManagedStorageAccountAttribute : KeyVaultCmdletBase
+    {
+        #region Input Parameter Definitions
+
+        [Parameter( Mandatory = true,
+            Position = 0,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "Vault name. Cmdlet constructs the FQDN of a vault based on the name and currently selected environment." )]
+        [ValidateNotNullOrEmpty]
+        public string VaultName { get; set; }
+
+        [Parameter( Mandatory = true,
+            Position = 1,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "Key Vault managed storage account name. Cmdlet constructs the FQDN of a managed storage account name from vault name, currently " +
+                          "selected environment and manged storage account name." )]
+        [ValidateNotNullOrEmpty]
+        [Alias( Constants.StorageAccountName )]
+        public string AccountName { get; set; }
+
+        [Parameter( Mandatory = false,
+            ValueFromPipelineByPropertyName = true,
```

Comment:
I don't think piping makes sense for these values for a PATCH scenario - the entire idea of PATCH is to just provide those proeprties that you want to change, and not bing the other properties.

Once again, the ```InputObject``` piping model from above would work here, to just allow piping the identifyign properties of the managed storage account, and not any of its behavioral properties.